File: frontend/src/types/product.ts
export interface ProductVariant {
  key: string;
  value: string;
  stock: number;
  price_override: number | null;
  compare_at_price_override: number | null;
  cost_price_override: number | null;
  weight_grams_override: number | null;
  height_cm_override: number | null;
  sku_override: string | null;
  active: boolean;
  recommended_pot_ids: string[];
}

export type ProductType = "simple" | "combo";

export interface ComboItem {
  product_id: string;
  quantity: number;
}

export interface ComboItemDetail extends ComboItem {
  title: string;
  image_url: string | null;
  price: number;
  stock: number;
}

export interface ProductSummary {
  product_id: string;
  tenant_id: string;
  tenant_name: string;
  title: string;
  short_description: string | null;
  price: number;
  compare_at_price: number | null;
  cost_price: number | null;
  currency: string;
  image_url: string | null;
  is_featured: boolean;
  rating_avg: number | null;
  rating_count: number;
  status: string;
  category_id: string | null;
  stock: number;
  tags: string[];
  care: Record<string, string>;
  product_type: ProductType;
}

/** Las tres plantillas de importación — no comparten propiedades (cuidados
 * de planta, color de maceta), por eso van separadas. */
export type ImportProductKind = "plantas" | "macetas" | "quimicos";

export const IMPORT_KIND_LABEL: Record<ImportProductKind, string> = {
  plantas: "Plantas",
  macetas: "Macetas y Accesorios",
  quimicos: "Productos Químicos",
};

export interface ImportJobRow {
  row: number;
  name: string | null;
  action: "created" | "updated" | "warning" | "error";
  message: string | null;
}

export interface ImportJob {
  job_id: string;
  status: "processing" | "completed" | "failed";
  filename: string | null;
  total: number;
  processed: number;
  created: number;
  updated: number;
  warnings: number;
  errors: number;
  rows: ImportJobRow[];
  error: string | null;
  created_at: string | null;
  finished_at: string | null;
}

export interface ImageSuggestionCandidate {
  title: string;
  thumbnail_url: string;
  source_url: string;
  license: string | null;
  attribution: string | null;
}

export interface ImageSuggestionResponse {
  query: string;
  candidates: ImageSuggestionCandidate[];
}

export interface BulkImageSuggestionRow {
  product_id: string;
  title: string;
  query: string;
  candidates: ImageSuggestionCandidate[];
  error: string | null;
}

export interface BulkConfirmResult {
  product_id: string;
  status: "ok" | "error";
  urls: string[];
  message: string | null;
}

export interface ProductDetail extends ProductSummary {
  description: string | null;
  target_markup_pct: number | null;
  images: string[];
  stock: number;
  variants: ProductVariant[];
  category_id: string | null;
  category_name: string | null;
  category_slug: string | null;
  sold_count: number;
  sku: string | null;
  tags: string[];
  tax: string;
  weight_grams: number | null;
  height_cm: number | null;
  care: Record<string, string>;
  attributes: Record<string, string>;
  recommended_pot_ids: string[];
  combo_items: ComboItemDetail[];
}

File: backend/app/schemas/product.py
from typing import Literal

from pydantic import BaseModel, Field


class VariantSchema(BaseModel):
    key: str
    value: str
    stock: int = 0
    price_override: int | None = None
    compare_at_price_override: int | None = None
    weight_grams_override: int | None = None
    height_cm_override: int | None = None
    sku_override: str | None = None
    active: bool = True
    recommended_pot_ids: list[str] = []


class ProductCreate(BaseModel):
    title: str = Field(max_length=150)
    short_description: str | None = Field(None, max_length=160)
    description: str | None = None
    price: int = Field(gt=0)
    compare_at_price: int | None = None
    currency: Literal["ARS", "USD"] = "ARS"
    tax: Literal["iva-21", "iva-10", "exento"] = "iva-21"
    category_id: str | None = None
    stock: int = Field(ge=0, default=0)
    sku: str | None = None
    is_featured: bool = False
    publish_at: str | None = None
    weight_grams: int | None = None
    height_cm: int | None = None
    tags: list[str] = []
    care: dict[str, str] = {}
    attributes: dict[str, str] = {}
    variants: list[VariantSchema] = []
    recommended_pot_ids: list[str] = []


class ProductUpdate(BaseModel):
    title: str | None = Field(None, max_length=150)
    short_description: str | None = Field(None, max_length=160)
    description: str | None = None
    price: int | None = Field(None, gt=0)
    compare_at_price: int | None = None
    currency: Literal["ARS", "USD"] | None = None
    tax: Literal["iva-21", "iva-10", "exento"] | None = None
    category_id: str | None = None
    stock: int | None = Field(None, ge=0)
    sku: str | None = None
    status: Literal["draft", "active", "paused"] | None = None
    is_featured: bool | None = None
    publish_at: str | None = None
    weight_grams: int | None = None
    height_cm: int | None = None
    tags: list[str] | None = None
    care: dict[str, str] | None = None
    attributes: dict[str, str] | None = None
    images: list[str] | None = None
    variants: list[VariantSchema] | None = None
    recommended_pot_ids: list[str] | None = None


class ProductSummary(BaseModel):
    product_id: str
    tenant_id: str
    tenant_name: str
    title: str
    short_description: str | None = None
    price: int
    compare_at_price: int | None = None
    currency: str = "ARS"
    image_url: str | None = None
    is_featured: bool = False
    rating_avg: float | None = None
    rating_count: int
    status: str
    category_id: str | None = None
    stock: int = 0
    tags: list[str] = []
    care: dict[str, str] = {}


class ProductDetail(ProductSummary):
    description: str | None = None
    images: list[str]
    stock: int
    variants: list[VariantSchema]
    category_id: str | None = None
    category_name: str | None = None
    sold_count: int
    sku: str | None = None
    tags: list[str]
    tax: str = "iva-21"
    weight_grams: int | None = None
    height_cm: int | None = None
    care: dict[str, str] = {}
    attributes: dict[str, str] = {}
    recommended_pot_ids: list[str] = []
